step-controller-plugins/step-controller-plugins-streaming/src/main/java/step/plugins/streaming/ContextAwareWebsocketUploadEndpoint.java

```diff
@@ -0,0 +1,59 @@
+package step.plugins.streaming;
+
+import jakarta.websocket.CloseReason;
+import jakarta.websocket.EndpointConfig;
+import jakarta.websocket.Session;
+import step.functions.handler.liveupload.LiveUploadContext;
+import step.functions.handler.liveupload.LiveUpload;
+import step.functions.handler.liveupload.LiveUploadContextListeners;
+import step.streaming.common.StreamingResourceMetadata;
+import step.streaming.common.StreamingResourceReference;
+import step.streaming.common.StreamingResourceStatus;
+import step.streaming.server.StreamingResourceManager;
+import step.streaming.websocket.server.WebsocketServerEndpointSessionsHandler;
+import step.streaming.websocket.server.WebsocketUploadEndpoint;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+public class ContextAwareWebsocketUploadEndpoint extends WebsocketUploadEndpoint {
+
+    public ContextAwareWebsocketUploadEndpoint(StreamingResourceManager manager, WebsocketServerEndpointSessionsHandler sessionsHandler) {
+        super(manager, sessionsHandler);
+    }
+
+    private String liveUploadContextId = null;
```

Comment:
Seeing that we store the resource metadata and references here, I'm asking me if we create one instance of this Endpoint for each uploaded resource. What is the lifecycle of this class?